src/components/Video/VideoContent/VideoContent.tsx

```diff
@@ -0,0 +1,190 @@
+import React, { useState, useEffect, useRef, useCallback } from 'react';
+import videojs from 'video.js';
+import Component from 'video.js/dist/types/component.d';
+import Player from 'video.js/dist/types/player.d';
+import classNames from 'classnames';
+import { useSearchParams } from 'next/navigation';
+import styles from './VideoContent.module.scss';
+import LazyLoad from '../../LazyLoad/LazyLoad';
+import useSharedUnmutedVideoState from '../useSharedUnmutedVideoState';
+import 'video.js/dist/video-js.css';
+import Controls from '../Controls/Controls';
+import { VideoData } from '../../../types/types';
+
+export type VideoContentProps = {
+  hasControls: boolean;
+  isAutoplaying: boolean;
+  isLooping: boolean;
+  video: VideoData;
+};
+
+type VideoJsComponent = Component & {
+  handleClick: () => void;
+};
+
+const VideoContent = ({
+  isAutoplaying,
+  hasControls,
+  isLooping,
+  video: { baseUrl, blur, height, hls, width },
+}: VideoContentProps) => {
+  const videoContainerRef = useRef(null);
+
+  const searchParams = useSearchParams();
+  const hasFocusMode = searchParams.get('hasFocusMode');
+
+  const [player, setPlayer] = useState<Player>(null);
+  const [isPlaying, setIsPlaying] = useState(isAutoplaying);
+  const [hasLoaded, setHasLoaded] = useState(false);
+
+  const [isMuted, toggleIsMuted] = useSharedUnmutedVideoState(hls ?? 'unknown');
+
+  const handlePlayToggle = () => {
+    if (!player) {
+      return;
+    }
+
+    if (player.paused()) {
+      player.play();
+
+      return;
+    }
+
+    player.pause();
+  };
+
+  const handleLoadVideo = useCallback(() => {
```

Comment:
Normally these follow a pattern of `handle[EventThatNeedsToBeHandled]`. This seems more like `handle[BehaviorOfTheHandler]`. `handleCameIntoView` or something similar, or renaming this `loadVideo` and calling it from an inline function in the `onIntersect` callback (I know functions defined inline cause rerenders bla bla, but doubt it leads to any measurable performance impact)